internal/server/v1beta1/search_test.go

```diff
@@ -332,3 +334,235 @@ func TestSuggest(t *testing.T) {
+				},
+			},
+			Setup: func(ctx context.Context, as *mocks.AssetService) {
+				cfg := asset.GroupConfig{
+					GroupBy: []string{"resource"},
+					Filters: map[string][]string{
+						"type":           {"topic"},
+						"service":        {"kafka", "rabbitmq"},
+						"data.landscape": {"th"},
+					},
+				}
+				as.EXPECT().GroupAssets(ctx, cfg).Return([]asset.GroupResult{}, nil)
+			},
+		},
+		{
+			Description: "should pass include fields to search config format",
+			Request: &compassv1beta1.GroupAssetsRequest{
+				Groupby: []string{"resource"},
+				Filter: map[string]string{
+					"data.landscape": "th",
+					"type":           "topic",
+					"service":        "kafka,rabbitmq",
+				},
+				IncludeFields: []string{"data.columns.name", "owners.email"},
+			},
+			Setup: func(ctx context.Context, as *mocks.AssetService) {
+				cfg := asset.GroupConfig{
+					GroupBy: []string{"resource"},
+					Filters: map[string][]string{
+						"type":           {"topic"},
+						"service":        {"kafka", "rabbitmq"},
+						"data.landscape": {"th"},
+					},
+					IncludedFields: []string{"data.columns.name", "owners.email"},
+				}
+				as.EXPECT().GroupAssets(ctx, cfg).Return([]asset.GroupResult{}, nil)
+			},
+		},
+		{
+			Description: "should return the grouped documents",
+			Request: &compassv1beta1.GroupAssetsRequest{
+				Groupby: []string{"resource"},
+			},
+			Setup: func(ctx context.Context, as *mocks.AssetService) {
+				cfg := asset.GroupConfig{
+					GroupBy: []string{"resource"},
+					Filters: make(map[string][]string),
+				}
+				response := []asset.GroupResult{
+					{
+						GroupFields: []asset.GroupField{{
+							GroupKey:   "resource",
+							GroupValue: "kafka",
+						},
+						},
+						Assets: []asset.Asset{{
+							Type:        "test",
+							ID:          "test-resource",
+							Description: "some description",
+							Service:     "test-service",
+							Labels: map[string]string{
+								"entity":    "gotocompany",
+								"landscape": "id",
+							},
+						},
+						},
+					},
+				}
+				as.EXPECT().GroupAssets(ctx, cfg).Return(response, nil)
+			},
+
+			PostCheck: func(resp *compassv1beta1.GroupAssetsResponse) error {
+				expected := &compassv1beta1.GroupAssetsResponse{
+					AssetGroups: []*compassv1beta1.AssetGroup{
+						{
+							GroupFields: []*compassv1beta1.GroupField{
+								{
+									GroupKey:   "resource",
+									GroupValue: "kafka",
+								},
+							},
+							Assets: []*compassv1beta1.Asset{
+								{
+									Id:          "test-resource",
+									Description: "some description",
+									Service:     "test-service",
+									Type:        "test",
+									Labels: map[string]string{
+										"entity":    "gotocompany",
+										"landscape": "id",
+									},
+								},
+							},
+						},
+					},
+				}
+				testutils.AssertEqualProto(t, expected, resp)
+				return nil
+			},
+		},
+		{
+			Description: "should return the requested number of assets",
```

Comment:
I think we can remove this test, the size aspect cannot be tested for in this layer.

Reply:
done